cloud-sql-proxy/cloud-sql-proxy.sh

```diff
@@ -214,6 +226,50 @@ EOF
 
 }
 
+
+# Helper to run any command with default bdutil settings with Fibonacci
+# backoff. If all retries fail, returns last attempt's exit code.
+# Args: "$@" is the command to run.
+function run_with_retries() {
+  local retry_backoff=(1 1 2 3 5 8 13 21 34 55 89 144)
+  local -a cmd=("$@")
+  echo "About to run '${cmd[*]}' with retries..."
+
+  local update_succeeded=0
+  for ((i = 0; i < ${#retry_backoff[@]}; i++)); do
+    if "${cmd[@]}"; then
+      update_succeeded=1
+      break
+    else
+      local sleep_time=${retry_backoff[$i]}
+      echo "'${cmd[*]}' attempt $(( $i + 1 )) failed! Sleeping ${sleep_time}." >&2
+      sleep ${sleep_time}
+    fi
+  done
+
+  if ! (( ${update_succeeded} )); then
+    echo "Final attempt of '${cmd[*]}'..."
+    # Let any final error propagate all the way out to any error traps.
+    "${cmd[@]}"
+  fi
+}
+
+
+function configure_hive_warehouse_dir(){
+  run_with_retries nc -zv "${MASTER_HOSTNAMES[0]}" 3306
```

Comment:
I'm not sure this is what you want. This will succeed at the very beginning of master 0's setup when mysql server is still running on 3306. You might actually just be able to run the next line with retries

Reply:
Good point. After a few test run, I managed to figure out the correct order, should run_validation until it succeeds before moving on